chainer/distributions/normal.py

```diff
@@ -32,14 +32,29 @@ class Normal(distribution.Distribution):
         location :math:`\\mu`. This is the mean parameter.
         scale(:class:`~chainer.Variable` or :class:`numpy.ndarray` or \
         :class:`cupy.ndarray`): Parameter of distribution representing the \
-        scale :math:`\\sigma`.
+        scale :math:`\\sigma`. Either `scale` or `log_var` (not both) must \
+        have a value.
+        log_var(:class:`~chainer.Variable` or :class:`numpy.ndarray` or \
+        :class:`cupy.ndarray`): Parameter of distribution representing the \
+        scale :math:`\\log(\\sigma^2)`. Either `scale` or `log_var` (not \
+        both) must have a value.
 
     """
 
-    def __init__(self, loc, scale):
+    def __init__(self, loc, scale=None, log_var=None):
         super(Normal, self).__init__()
+        if not (scale is None) ^ (log_var is None):
+            raise ValueError(
+                "Either `scale` or `log_var` (not both) must have a value.")
         self.loc = chainer.as_variable(loc)
-        self.scale = chainer.as_variable(scale)
+
+        with chainer.using_config('enable_backprop', True):
+            if scale is None:
+                self.log_var = chainer.as_variable(log_var)
+                self.scale = exponential.exp(0.5 * self.log_var)
+            else:
+                self.scale = chainer.as_variable(scale)
+                self.log_var = 2. * exponential.log(self.scale)
```

Comment:
Could you make `scale` and `log_var` read-only, or define setters of them to do this always?

BTW, it seems not bad to make all attributes of `Distribution` subclasses read-only.